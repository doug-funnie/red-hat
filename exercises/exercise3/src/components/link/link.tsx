import glamorous, { CSSProperties } from 'glamorous';
import { lighten } from 'polished';
import * as React from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { ThemedProps } from 'styles/theme';

const linkStyles: CSSProperties = {
  fontWeight: 700,
  textDecoration: 'none',
};

const getLinkStylesFromTheme = ({ theme }: ThemedProps) => ({
  color: theme.colors.red,
  ':hover': {
    color: lighten(0.2, theme.colors.red),
  },
});

export const ExternalLink = glamorous.a(linkStyles, getLinkStylesFromTheme);
export const InternalLink = glamorous(RouterLink)(linkStyles, getLinkStylesFromTheme);

export interface Props {
  to: string;
  isExternal?: boolean;
  css?: CSSProperties;
}

export const Link: React.SFC<Props> = ({ to, isExternal, children, css }) => isExternal
  ? (
    <ExternalLink
      href={to}
      rel='noopener noreferer'
      target='_blank'
      css={css}
    >
      {children}
    </ExternalLink>
  ) : (
    <InternalLink to={to} css={css}>
      {children}
    </InternalLink>
  );
