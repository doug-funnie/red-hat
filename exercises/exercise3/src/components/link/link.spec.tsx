import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { Link, Props } from './link';

const props: Props = {
  to: 'to',
};

it('renders internal link by default', () => {
  const view = shallow(<Link {...props} />);
  expect(toJson(view)).toMatchSnapshot();
});

it('renders external link if isExternal is true', () => {
  const view = shallow(<Link {...props} isExternal />);
  expect(toJson(view)).toMatchSnapshot();
});
