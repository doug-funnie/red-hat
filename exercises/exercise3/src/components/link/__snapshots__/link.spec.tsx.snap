// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders external link if isExternal is true 1`] = `
<glamorous(a)
  href="to"
  rel="noopener noreferer"
  target="_blank"
/>
`;

exports[`renders internal link by default 1`] = `
<glamorous(Link)
  to="to"
/>
`;
